polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,283 @@
+        dset.attrs["end_time"] = ds_attrs["end_time"].isoformat()
+
+        return
+
+    def save_datasets(
+        self,
+        dataset: list[xr.DataArray],
+        filename=None,
+        dtype=None,
+        append=True,
+        compute=True,
+        chunks=None,
+        **kwargs,
+    ):
+        """Save hdf5 datasets."""
+
+        compression = kwargs.pop("compression", None) if "compression" in kwargs else None
+        if compression == "none":
+            compression = None
+
+        add_geolocation = kwargs.pop("add_geolocation") if "add_geolocation" in kwargs else False
+
+        # will this be written to one or multiple files?
+        output_names = []
+
+        for dataset_id in dataset:
+            args = self._output_file_kwargs(dataset_id, dtype)
+            out_filename = filename or self.get_filename(**args)
+            output_names.append(out_filename)
+        one_file = all_equal(output_names)
+
+        filename = output_names[0]
+        if not all_equal(output_names):
+            LOG.warning("Writing to {}".format(filename))
+
+        hdf5_fh = self.open_hdf5_filehandle(filename, append=append)
+
+        datasets_by_area = self.iter_by_area(dataset)
+        # Initialize source/targets at start of each new AREA grouping.
+        dsets = []
+        targets = []
+
+        for area, data_arrs in datasets_by_area:
+            # open hdf5 file handle, check if group already exists.
+            parent_group = self.create_proj_group(filename, hdf5_fh, area)
+
+            if add_geolocation:
+                chunks = data_arrs[0].chunks
+                geo_sets, fnames = self.write_geolocation(
+                    hdf5_fh, filename, parent_group, area, dtype, append, compression, chunks
+                )
+                dsets.append(geo_sets)
+                targets.append(fnames)
+
+            for data_arr in data_arrs:
+                d_dtype = data_arr.dtype if dtype is None else dtype
+                hdf_subgroup = "{}/{}".format(parent_group, data_arr.attrs["name"])
+
+                file_var = FakeHDF5(filename, hdf_subgroup)
+                arr_dset = self.check_variable(hdf5_fh, hdf_subgroup, data_arr.data, d_dtype, append, compression)
+                dsets.append(data_arr.data)
+                targets.append(file_var)
+
+        results = (dsets, targets)
+        if compute:
+            LOG.info("Computing and writing results...")
+            return compute_writer_results([results])
+
+        targets, sources, delayeds = split_results([results])
+        if delayeds:
+            # This writer had only delayed writes
+            return delayeds
+        else:
+            return targets, sources
+
+
+def add_writer_argument_groups(parser, group=None):
+    """Create writer argument groups."""
+    from argparse import SUPPRESS
+
+    if group is None:
+        group = parser.add_argument_group(title="hdf5 Writer")
+    group.add_argument(
+        "--output-pattern",
```

Comment:
I think to match the new writers this should be `--output-filename`.

Reply:
There will be no backwards compatibility with this?  If someone enters --output-pattern, it should fail?

Reply:
Or should I be nice and convert it.

Reply:
I was thinking no there wouldn't be backwards compatibility, but I'm not sure what @kathys and I decided on this in one of our recent meetings. If you try to convert to the new kwarg it would have to be handled by the glue script and not the writer as we'd have to do it for Satpy-centric writers too.

Reply:
@kathys @djhoese Not sure of your opinion, the argument parser will throw an error which is pretty clear about --output-pattern being the wrong argument.  What it can't do at the moment is redirect the user to using the --output-filename flag.  However, this could get ugly over time if multiple old flags need to throw individualized messages.  Either, leaving with the simple error message or printing the default help from the parser might be the better than trying anything more customized.  @djhose, your opinion?  Am I walking down a path that is too hard to support?

Reply:
Similar to have the `type=` defined for the argument as a function that does the legacy keys -> new keys conversion, we could have the function print a message telling them the new flag to use *or* raise an error telling them the new flag to use. The help message could also be updated to mention the new flag.

Is any of this necessary...I don't know. Is it easier to just make users learn the new way? Maybe?

Reply:
I am going to back track on this, if error doesn't get raised early, argparse doesn't actually throw the error for output-patter, because it actually does exist.  So it is better to just let argparse throw and error and then let the user go back to the docs. 
```
usage: 
    /Users/joleenf/code/polar2grid/polar2grid/glue.py -h
see available products:
    /Users/joleenf/code/polar2grid/polar2grid/glue.py -r <reader> -w <writer> --list-products -f file1 [file2 ...]
basic processing:
    /Users/joleenf/code/polar2grid/polar2grid/glue.py -r <reader> -w <writer> [options] -f file1 [file2 ...]
basic processing with limited products:
    /Users/joleenf/code/polar2grid/polar2grid/glue.py -r <reader> -w <writer> [options] -p prod1 prod2 -f file1 [file2 ...]
/Users/joleenf/code/polar2grid/polar2grid/glue.py: error: unrecognized arguments: --output-pattern

```